src/main/kotlin/me/zeroeightsix/kami/feature/module/Nuker.kt

```diff
@@ -0,0 +1,132 @@
+            }
+
+            progress += state.calcBlockBreakingDelta(it.player, it.world, currentBlock)
+            if (progress >= 1.0) {
+                stopMine(currentBlock!!)
+                progress = 0.0
+                currentBlock = null
+            }
+        }
+    })
+
+    private var progress = 0.0
+    private var currentBlock: BlockPos? = null
+
+    private fun getBoxCorner(playerPos: Vec3d, negative: Boolean = false) =
+        playerPos.add(Vec3d(hitRange, hitRange, hitRange).run { if (negative) negate() else this })
+
+    private fun startMine(blockPos: BlockPos) {
+        mc.networkHandler?.sendPacket(
+            PlayerActionC2SPacket(
+                PlayerActionC2SPacket.Action.START_DESTROY_BLOCK,
+                blockPos,
+                Direction.DOWN
+            )
+        )
+    }
+
+    private fun stopMine(blockPos: BlockPos) {
+        mc.networkHandler?.sendPacket(
+            PlayerActionC2SPacket(
+                PlayerActionC2SPacket.Action.STOP_DESTROY_BLOCK,
+                blockPos,
+                Direction.DOWN
+            )
+        )
+    }
+
+    private fun nextBlock(player: ClientPlayerEntity, world: ClientWorld) {
+        for (block in BlockPos.method_29715(Box(getBoxCorner(player.pos), getBoxCorner(player.pos, true)))) {
+            if (validate(player, world, block)) {
+                currentBlock = block
+                break
+            }
+        }
+    }
+
+    private fun instantMineBlocks(player: ClientPlayerEntity, world: ClientWorld) {
+        for (block in BlockPos.method_29715(Box(getBoxCorner(player.pos), getBoxCorner(player.pos, true)))) {
+            if (validate(player, world, block) && canInstaMine(player, world, block)) {
+                startMine(block)
+            }
+        }
+    }
+
+    private fun canInstaMine(player: ClientPlayerEntity, world: ClientWorld, block: BlockPos) =
+        player.isCreative || world.getBlockState(block).calcBlockBreakingDelta(player, world, block) >= 1
+
+    private fun validate(player: ClientPlayerEntity, world: ClientWorld, block: BlockPos): Boolean {
+        val state = world.getBlockState(block)
+
+        val throughBlockCheck = throughBlocks ||
+            world.raycast(
+                RaycastContext(
+                    Vec3d(player.pos.x, player.eyeY, player.pos.z),
+                    block.asVec3d.add(single_vec(0.5)),
+                    RaycastContext.ShapeType.COLLIDER,
+                    RaycastContext.FluidHandling.NONE,
+                    player
+                )
+            ).blockPos == block
+
+        val blockCheck = !state.isAir &&
+            state.block !is FluidBlock &&
+            block.isWithinDistance(
+                player.pos,
+                hitRange
+            ) &&
+            // this stupid way of checking for unbreakable blocks was not my idea
+            (player.isCreative || !(block is CommandBlock || block is StructureBlock || block is JigsawBlock))
```

Comment:
Looks like unbreakable blocks have a hardness of `-1.0f` (check `Blocks.java` (minecraft class)'s definitions.) Try checking the hardness instead of this?